hack/make/build-deb

```diff
@@ -86,6 +86,16 @@ set -e
 		cat >> "$DEST/$version/Dockerfile.build" <<-EOF
 			RUN cp -aL hack/make/.build-deb debian
 			RUN { echo '$debSource (${debVersion}-0~${suite}) $suite; urgency=low'; echo; echo '  * Version: $VERSION'; echo; echo " -- $debMaintainer  $debDate"; } > debian/changelog && cat >&2 debian/changelog
+		EOF
+		# Remove the following case-based substitution when none of these are supported, and the TasksMax value is uncommented in docker.service
+		case "$version" in
+		debian-jessie|debian-wheezy|ubuntu-precise|ubuntu-trusty|ubuntu-wily)
+			;;
+		*)
+			echo "RUN sed -i -- 's/#TasksMax=infinity/TasksMax=infinity/' contrib/init/systemd/docker.service" >> "$DEST/$version/Dockerfile.build"
```

Comment:
I also thought this should be "deep" enough. But the git commit check seems to run afterwards.

This is the relevant output of a `make deb` building current rc1 on armhf:

```
Step 14 : RUN sed -i -- 's/#TasksMax=infinity/TasksMax=infinity/' contrib/init/systemd/docker.service
 ---> Running in 708e5b0729ec
 ---> fd894af223c8
Removing intermediate container 708e5b0729ec
Step 15 : RUN dpkg-buildpackage -uc -us -I.git
 ---> Running in eb84d00712e7
dpkg-buildpackage: source package docker-engine
dpkg-buildpackage: source version 1.12.0~rc1-0~jessie
dpkg-buildpackage: source distribution jessie
dpkg-buildpackage: source changed by Docker <support@docker.com>
 dpkg-source -I.git --before-build docker
dpkg-buildpackage: host architecture armhf
 debian/rules clean
dh clean --with=bash-completion --with=systemd
   dh_testdir
   dh_auto_clean
   dh_clean
 dpkg-source -I.git -b docker
dpkg-source: warning: no source format specified in debian/source/format, see dpkg-source(1)
dpkg-source: warning: source directory 'docker' is not <sourcepackage>-<upstreamversion> 'docker-engine-1.12.0~rc1'
dpkg-source: info: using source format `1.0'
dpkg-source: info: building docker-engine in docker-engine_1.12.0~rc1-0~jessie.tar.gz
dpkg-source: info: building docker-engine in docker-engine_1.12.0~rc1-0~jessie.dsc
 debian/rules build
dh build --with=bash-completion --with=systemd
   dh_testdir
   dh_auto_configure
   debian/rules override_dh_auto_build
make[1]: Entering directory '/usr/src/docker'
./hack/make.sh dynbinary

# WARNING! I don't seem to be running in a Docker container.
# The result of this command might be an incorrect build, and will not be
# officially supported.
#
# Try this instead: make all
#
#~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# GITCOMMIT = 1f136c1-unsupported
# The version you are building is listed as unsupported because
# there are some files in the git repository that are in an uncommited state.
# Commit these changes, or add to .gitignore to remove the -unsupported from the version.
# Here is the current list:
 M contrib/init/systemd/docker.service
#~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
bundles/1.12.0-rc1 already exists. Removing.
```